proxmox/resource_cloud_init.go

```diff
@@ -0,0 +1,174 @@
+package proxmox
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/Telmate/proxmox-api-go/proxmox"
+	"github.com/kdomanski/iso9660"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	isoContentType = "iso"
+)
+
+func resourceCloudInitDisk() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudInitDiskCreate,
+		ReadContext:   resourceCloudInitDiskRead,
+		DeleteContext: resourceCloudInitDiskDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"pve_node": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"storage": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"user_data": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"meta_data": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"network_config": {
```

Comment:
I would add "vendor_data" field https://canonical-cloud-init.readthedocs-hosted.com/en/latest/explanation/vendordata.html

It would be exactly the same as others but run only at first boot, proxmox support it too